sklearn/preprocessing/_function_transformer.py

```diff
@@ -444,3 +464,14 @@ def _sk_visual_block_(self):
             name_caption="FunctionTransformer",
             doc_link_label="FunctionTransformer",
         )
+
+    def _set_dataframe_feature_names(self, X):
+        """Get output column names from `func` output."""
+        if _is_polars_df(X) or _is_pandas_df(X):
+            head = X.head(1)
+        else:
+            head = X[:1]
+
+        head_out = self.func(head)
+        if _is_polars_df(head_out) or _is_pandas_df(head_out):
```

Comment:
we don't need to check for pandas or polars, do we? It's done in `_get_feature_names`